xpublish_wms/wms/get_map.py

```diff
@@ -235,23 +238,34 @@ def select_elevation(self, ds: xr.Dataset, da: xr.DataArray) -> xr.DataArray:
     def select_custom_dim(self, da: xr.DataArray) -> xr.DataArray:
         """
         Select other dimension, ensuring a 2D array
+
+        If dimension is provided :
+            - use xarray to access custom coord
+            - method nearest to ensure at least one result
+
+        Otherwise:
+            - Get first value of coord
+
         :param da:
         :return:
         """
         # Filter dimension from custom query, if any
         for dim, value in self.dim_selectors.items():
             if dim in da.coords:
-                dtype = da[dim].dtype
-                method = None
-                if "timedelta" in str(dtype):
-                    value = pd.to_timedelta(value)
-                elif np.issubdtype(dtype, np.integer):
-                    value = int(value)
-                    method = "nearest"
-                elif np.issubdtype(dtype, np.floating):
-                    value = float(value)
-                    method = "nearest"
-                da = da.sel({dim: value}, method=method)
+                if value is None:
+                    da = da.isel({dim: 0})
```

Comment:
I am guessing this is necessary because otherwise you get an error if you dont specify?

Reply:
well the actual issue was happening at L177 where it would try to get a query parameter that could sometimes be missing (as it seems like we allow any extent to be missing from the query), so that's why I added the `if k in query_params else None` there. And since I would imagine it wouldn't work to try and get the closest value to `None`, I added that condition to just get the first value.

Reply:
oooook got it. Ok so i think L177 might work as this instead? This would filter instead of having to set them as None

```
{ k: query_params[k] for k in available_selectors if k in query_params}
``

Reply:
That would work except that we would need to use `ds.gridded.additional_coords(ds[self.parameter])` to iterate through the extra dimensions rather than `self.dim_selectors`, which I'm not sure about. We would always need to filter the data by those dims in order for to have the correct number of dims for `GetMap` right?

Reply:
Yeah its true. Its a decision about whether a tile should work by default vs if it should  fail if the user didnt pass in dimensions. 

OKay I like your way, it matcehs the get caps too!